src/wp-includes/rest-api/endpoints/class-wp-rest-templates-controller.php

```diff
@@ -269,15 +270,14 @@ public function update_item( $request ) {
 			return $fields_update;
 		}
 
-		$post = get_post( $template->wp_id );
-		wp_after_insert_post( $post, $update, $post_before );
-
 		$request->set_param( 'context', 'edit' );
 
 		$post = get_post( $template->wp_id );
 		/** This action is documented in wp-includes/rest-api/endpoints/class-wp-rest-posts-controller.php */
 		do_action( "rest_after_insert_{$this->post_type}", $post, $request, false );
 
+		wp_after_insert_post( $post, $update, $post_before );
+
```

Comment:
Great catch moving this to follow the `rest_after_insert_post_type` action: I totally missed that inconsistency.